src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case "addProperties":
+        currentState = Object.assign({}, currentState, action.extraData);
```

Comment:
we don't have to reassign currentState
```suggestion
        Object.assign(currentState, action.extraData);
```